packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+import type { StandardMaterial } from "./standardMaterial";
+import { RegisterClass } from "../Misc/typeStore";
+import { Color3, Vector3 } from "core/Maths/math";
+import type { Mesh } from "core/Meshes/mesh";
+import { Logger } from "core/Misc/logger";
+import { expandToProperty, serialize } from "core/Misc/decorators";
+import type { AbstractMesh } from "core/Meshes/abstractMesh";
+
+/**
+ * Supported visualizations of MeshDebugPluginMaterial
+ */
+export enum MeshDebugMode {
+    /**
+     * Material without any mesh debug visualization
+     */
+    NONE = -1,
+    /**
+     * A wireframe of the mesh
+     * NOTE: For this mode to work correctly, bar() must first be called on mesh.
+     */
+    TRIANGLES = 0,
+    /**
+     * A wireframe of the mesh, with dots drawn over vertices
+     * NOTE: For this mode to work correctly, foo() must first be called on mesh.
+     */
+    VERTICES = 1,
+    /**
+     * A checkerboard grid of the mesh's UV set 0
+     */
+    UV0 = 2,
+    /**
+     * A checkerboard grid of the mesh's UV set 1
+     */
+    UV1 = 3,
+    /**
+     * The mesh's vertex colors displayed as the primary texture
+     */
+    VERTEXCOLORS = 4,
+    /**
+     * An arbitrary, distinguishable color to identify the material
+     */
+    MATERIALIDS = 5,
+}
+
+/**
+ * Options for MeshDebugPluginMaterial that are given at initialization
+ */
+export interface MeshDebugOptions {
+    /**
+     * Mesh debug mode to initialize with.
+     * Defaults to NONE.
+     */
+    mode?: MeshDebugMode;
+    /**
+     * Whether the mesh debug visualization should multiply with any PBR debug colors underneath.
+     * Defaults to true.
+     */
+    multiplyDebug?: boolean;
+    /**
+     * Default diffuse color of the visualization.
+     * Defaults to (1.0, 1.0, 1.0).
+     */
+    shadedDiffuseColor?: Vector3;
+    /**
+     * Default specular color of the visualization.
+     * Defaults to (0.8, 0.8, 0.8).
+     */
+    shadedSpecularColor?: Vector3;
+    /**
+     * How dark the shading of the visualization is.
+     * Defaults to 10.
+     */
+    shadedSpecularPower?: number;
+    /**
+     * Width of edge lines in TRIANGLES and VERTICES modes.
+     * Defaults to 0.7.
+     */
+    wireframeThickness?: number;
+    /**
+     * Color of edge lines in TRIANGLES mode, and color of dots in VERTICES mode.
+     * Defaults to (0.0, 0.0, 0.0).
+     */
+    wireframeColor1?: Vector3;
+    /**
+     * Color of edge lines in VERTICES modes.
+     * Defaults to (0.9, 0.9, 0.9).
+     */
+    wireframeColor2?: Vector3;
+    /**
+     * Radius of dots drawn over vertices in VERTICES mode.
+     * Defaults to 1.2.
+     */
+    vertexRadius?: number;
+    /**
+     * Size of tiles in UV1 or UV2 modes.
+     * Defaults to 20.
+     */
+    uvScale?: number;
+    /**
+     * 1st color of checkerboard grid in UV1 or UV2 modes.
+     * Defaults to (1.0, 1.0, 1.0).
+     */
+    uvColor1?: Vector3;

```

Comment:
Should this not cover all UVs ?

Reply:
I wonder how it could be renamed in this case ? maybe uvPrimaryColor, uvSecondaryColor ?